src/status_backend/core.cljs

```diff
@@ -0,0 +1,80 @@
+(ns status-backend.core
+  (:require
+   [status-backend.config :as config]
+   [utils.network.core :as network]
+   [clojure.string :as string]
+   [oops.core :as oops]
+   [taoensso.timbre :as log]))
+
+(defn fetch
+  ([endpoint]
+   (fetch endpoint nil (fn [])))
+  ([endpoint body-params]
+   (fetch endpoint body-params (fn [])))
+  ([endpoint body-params callback]
+   (let [url    (str config/status-go-url endpoint)
+         params (assoc config/fetch-default-params :body body-params)]
+     (network/fetch url params callback))))
+
+(defn- method-name->endpoint [s]
+  (let [first-letter-capitalized (string/capitalize (first s))]
+    (str first-letter-capitalized (subs s 1))))
+
+(def fetch-js-obj
+  "Performs a request to the status-backend (by using `status-backend.core/fetch`) using
+  the method invoked as endpoint name. E.g
+  This call: (.myMethod fetch-js-obj params)
+  Is the same as: (fetch \"MyMethod\" params)
+  "
+  (js/Proxy. #js{}
+             #js{:get (fn [_target native-module-method-name]
+                        (let [endpoint (method-name->endpoint native-module-method-name)]
```

Comment:
Hello @qfrank 
Thank you for your review!

> this won't work ... We cannot simply determine the endpoint name based on the native-module-method-name because there are many unsupported endpoints in the status-backend server

I know there are unsupported endpoints, I'm not expecting this draft PR fully works right now, but this could be a pattern to follow.

Regarding to:

> For some unsupported endpoints, we need to use the corresponding V2 versions.
> I also created another https://github.com/status-im/status-mobile/pull/21450(not related to status backend integration) to use the V2 endpoint so that we can remove the old endpoint in the future


I don't fully agree with the approach taken. If we cannot infer the endpoint name based on the native module method name it implies:

1. We need to **manually** maintain a map (native-method -> endpoint) which will be error prone and will require an extra step from everyone while integrating a new native method. Since status-backend won't be used in production, we should reduce the divergence to the minimum.
2. Lack of standarization/consistency: Why do we want to give different names to functions that should only be wrappers?
3. As said in `2`, IMO, we should keep native modules related to status-go as thin as possible, so I don't agree with the approach taken in:
   - https://github.com/status-im/status-mobile/pull/21450
 
   Since it exposes a native method signature that is different from the V2 endpoint called inside. By doing this, we add an extra layer of abstraction/opacity from a call in CLJS to an execution in Status-go. Again, IMO, that PR should update the native methods signature and fix the usages in the CLJS side (i.e. call these native methods with a single param being a JS object).
   

Another option might be to integrate the status-backend calls inside the native module methods, so it'd be implementing the integration in Java/Kotlin, but think that'd be harder to maintain.

CC: @ilmotta

Reply:
@ulisesmac, I'll share my perspective, which can be incomplete.

> We need to manually maintain a map (native-method -> endpoint) which will be error prone

I was imagining this manual mapping will only exist temporarily, until status-backend has no unsupported endpoints or very few (which would be easy to handle with a blocklist).

> Lack of standarization/consistency: Why do we want to give different names to functions that should only be wrappers?

Isn't the case that unsupported endpoints always use the equivalent name, but with a `V2` suffix? This is a temporary situation, but there is some consistency.

> As said in 2, IMO, we should keep native modules related to status-go as thin as possible, so I don't agree with the approach taken in:

In the PR https://github.com/status-im/status-mobile/pull/21450, I understood that @qfrank opted to change the native implementation instead of the Clojure layer because  all the changes to build up the JSON request and call the v2 functions would be isolated as internal implementation, and no Clojure code had to be changed in that PR. If, in that PR, we change the native function signatures to take a JSON payload, we will need to change all CLJS calls as well. Maybe it was just a case of different devs with different tech skills and what was most comfortable.

But I have to agree with you @ulisesmac, our agreement is that we only write native code if truly necessary (e.g. performance), otherwise we use Clojure, which is easier to manipulate for most CCs and works well with hot-reload and the REPL.

Reply:
@qfrank 

 I agree on having a V2 version with a different API, for sure it is an improvement.
 
What I suggest is:
- Update the native methods' signature to match the V2 endpoints.
- Update the CLJS calls to these native methods (I can help with it, shouldn't be hard).
- Integrate Status backend as suggested in this PR, by just adding a conditional call.

However, if we prefer to have a native implementation then it's fine for me. IMO implementing the request mechanism in Kotlin and another in ObjC is harder to test and maintain. Not sure, but according to the new arch docs, we could (when we adopt it) implement it in C++ and share it across platforms.

Reply:
> I want this feature ASAP as I feel painful with build experience! Perhaps we can simply implement requesting the backend server in a native way that does not require much modification to ClojureScript. Only the configuration (server address/if enabled) needs to be passed to the native code. I can do it based on https://github.com/status-im/status-mobile/pull/21450 , WDYT? @ulisesmac @ilmotta

@qfrank @ulisesmac This is all highly technical work, with no impact to users, therefore, we can first try to make things work, even if they are kind of dirty, if the code is safe to merge of course. On the other hand, unfortunately every time we change these parts of the code we risk introducing regressions, and the cost of asking a QA to review such PRs is considerable because they have to check a lot of stuff, so let's keep in mind the cost of follow-up PRs touching so many endpoints.

But whatever you guys align I'll probably agree because, in the end, I also just want this integration to work first, so we can all profit from it even during the next 2.32 release.